cleanlab/classification.py

```diff
@@ -379,17 +452,21 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
                     "Assigning sample weights for final training based on estimated label quality."
                 )
 
-            self.sample_weight = np.ones(np.shape(labels_cleaned))
+            sample_weight = np.ones(np.shape(labels_cleaned))
             for k in range(self.num_classes):
                 sample_weight_k = 1.0 / max(self.noise_matrix[k][k], 1e-3)  # clip sample weights
-                self.sample_weight[labels_cleaned == k] = sample_weight_k
+                sample_weight[labels_cleaned == k] = sample_weight_k
 
+            sample_weight_expanded = np.zeros(len(labels))  # pad pruned examples with zero weight
+            sample_weight_expanded[x_mask] = sample_weight
+            self.label_issues_df["sample_weight"] = sample_weight_expanded
+            self.sample_weight = self.label_issues_df["sample_weight"]
             if self.verbose:
                 print("Fitting final model on the clean data ...")
             self.clf.fit(
                 x_cleaned,
                 labels_cleaned,
-                sample_weight=self.sample_weight,
+                sample_weight=sample_weight,
```

Comment:
a comment in the code explaining that `sample_weight` is length of the pruned data after cleaning out label issues and `self.sample_weight` is length of the entire dataset, could be helpful for developers (they're named the same but they are different... pretty tricky stuff with out a comment)

Reply:
added comments